onnxruntime/core/optimizer/insert_cast_transformer.cc

```diff
@@ -84,9 +84,7 @@ static bool NodeNeedsInputCastToFp32(const onnxruntime::Node& node) {
 // going to a node that will need a Cast.
 //
 // Return true if all the fp16 inputs and outputs are connected to nodes that will be cast to fp32.
-static bool IsIsolatedFp16NodeOnCpu(const onnxruntime::Node& node, onnxruntime::Graph& graph) {
-  bool isolated_fp16_node = false;
-
+static bool IsIsolatedFp16NodeOnCpu(const onnxruntime::Node& node, onnxruntime::Graph& graph, const KernelRegistry& cpu_kernel_registry) {
```

Comment:
Do we really need this check at all? If CPU has fp16 kernel for the node, why not just let it run?

Reply:
@skottmckay could you comment on this?